tests/utils/test_gen_coco_sal.py

```diff
@@ -1,67 +1,87 @@
         """
 
-        test_dets = kwcoco.CocoDataset(dets_file)
-        test_detector = TestDetector()
-        test_perturber = TestPerturber()
-        test_sal_gen = TestSalGenerator()
+        if is_usable:
+            real_import = builtins.__import__
 
-        sal_maps = gen_coco_sal(test_dets, test_detector, test_perturber, test_sal_gen, verbose=True)
+            # mock import function that acts as if kwcoco is not installed
+            def mock_import(name: str, *args: Any, **kw: Any) -> None:
+                if name == 'kwcoco':
+                    raise ModuleNotFoundError
+                return real_import(name, *args, **kw)
 
-        maps_file = os.path.join(DATA_DIR, 'coco_sal_maps.npy')
-        exp_sal_maps = np.load(maps_file, allow_pickle=True)[()]
+            # monkeypatch import function
+            builtins.__import__ = mock_import
 
-        assert sal_maps.keys() == exp_sal_maps.keys()
-        assert all([np.allclose(sal_maps[det_id], exp_sal_maps[det_id]) for det_id in sal_maps])
+            del sys.modules['xaitk_saliency.utils.gen_coco_sal']
 
-    def test_mismatched_labels(self) -> None:
-        """
-        Test that an exception is raised when test detections and detector have
-        differing class labels.
-        """
+        with pytest.raises(ImportError):
+            from xaitk_saliency.utils.gen_coco_sal import parse_coco_dset  # noqa: F401
 
-        test_dets = kwcoco.CocoDataset(dets_file)
-        test_detector = MismatchedDetector()
-        test_perturber = TestPerturber()
-        test_sal_gen = TestSalGenerator()
 
-        with pytest.raises(MismatchedLabelsError):
-            gen_coco_sal(test_dets, test_detector, test_perturber, test_sal_gen)
+@pytest.mark.skipif(not is_usable, reason="Extra 'xaitk-saliency[tools]' not installed.")
+class TestParseCocoDset:
 
-    def test_mismatched_name_labels(self) -> None:
+    def test_dset_parse(self) -> None:
         """
-        Test that an exception is raised when test detections and detector have
-        differing string class labels.
+        Test that a dummy detection file is parsed correctly.
         """
-        test_dets = kwcoco.CocoDataset(dets_file)
-        test_detector = MismatchedNameDetector()
-        test_perturber = TestPerturber()
-        test_sal_gen = TestSalGenerator()
-
-        with pytest.raises(MismatchedLabelsError):
-            gen_coco_sal(test_dets, test_detector, test_perturber, test_sal_gen)
+        dets_file = os.path.join(DATA_DIR, 'test_dets.json')
+        test_dset = kwcoco.CocoDataset(dets_file)
+
+        test_img_files = ['test_image1.png', 'test_image2.png']
+        test_imgs = [np.array(Image.open(os.path.join(DATA_DIR, f))) for f in test_img_files]
+
+        test_img_aids = [test_dset.gid_to_aids[gid] for gid in [1, 3]]
+
+        # construct expected bounding box and score matrices
+        test_bboxes = []
+        test_scores = []
+        for aids in test_img_aids:
```

Comment:
This is more or less replicating the logic in the actual code being tested, so this isn't a great test I think. In this kind of situation we should generally manually construct the data structure as we would expect the result to be and inject as little logic as possible unless there really does need to be something dynamic about it. If this data-structure is very large, we may want to pickle a serialization of it to load here (I don't think that's the case right now? ).